libssh-rs/src/channel.rs

```diff
@@ -44,6 +44,7 @@ impl Drop for Channel {
     fn drop(&mut self) {
         let (_sess, chan) = self.lock_session();
         unsafe {
+            sys::ssh_remove_channel_callbacks(chan, self._callbacks.as_ref() as *const _ as *mut _);
             sys::ssh_channel_free(chan);
         }
```

Comment:
I think this is slightly "better" in the sense that it is not casting a const ptr to a mut ptr:

```suggestion
        unsafe {
            // Prevent any callbacks firing as part the remainder of this drop operation
            sys::ssh_remove_channel_callbacks(self.chan_inner, self._callbacks.as_mut());
        }
        let (_sess, chan) = self.lock_session();
        unsafe {
            sys::ssh_channel_free(chan);
        }
```